pinot-transport/src/main/java/com/linkedin/pinot/transport/common/KeyedFuture.java

```diff
@@ -39,10 +40,9 @@
  * request dispatch with speculative (duplicate) request dispatching.
  *
  *
- * @param <K> Key type to identify a result/error
  * @param <V> Response Type
  */
-public interface KeyedFuture<K, V> extends ListenableFuture<Map<K, V>> {
+public interface KeyedFuture<V> extends ListenableFuture<Map<ServerInstance, V>> {
```

Comment:
I think this should be generic <k,v> given the name KeyedFuture. It can have wider use.

Better option is to have a thin derived impl ServerResponseFuture<V> (or something similar) that use serverInstance  as key